### PR TITLE
Lambda Workflows [builds lambda containers for operators] [4/5]

### PR DESCRIPTION
**Pull request by hsubbaraj-spiral**
## Describe your changes and why you are making these changes
Builds the docker images that lambda uses to run Aqueduct operators. There are some changes included to aqueduct_executor to enable lambda to work as well.
## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/dockerfiles/lambda/connectors/athena.dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+COPY lambda/connectors/data.py ${LAMBDA_TASK_ROOT}
+
+# Install the function's dependencies using file requirements.txt
+# from your project folder.
+
+COPY lambda/requirements.txt  .
+RUN pip3 install -r requirements.txt --target "${LAMBDA_TASK_ROOT}"
+RUN pip3 install awswrangler --target "${LAMBDA_TASK_ROOT}"
+
+RUN pip3 install --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple/ aqueduct-ml-test-hari==0.0.1 --target "${LAMBDA_TASK_ROOT}"
```
Comment (hsubbaraj-spiral) [inline src/dockerfiles/lambda/connectors/athena.dockerfile:13]: Change this to aqueduct-ml once the latest changes in this PR are cut in a release.

Comment (likawind) [reply src/dockerfiles/lambda/connectors/athena.dockerfile:13]: Will we hard-code release version here? This doc may need to update: https://www.notion.so/aqueducthq/Runbook-Cutting-an-OSS-Release-b72400b7584845f0b0b4ff84bab4aedc (@cw75 )

File: src/dockerfiles/lambda/function/function.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import base64
```
Comment (likawind) [inline src/dockerfiles/lambda/function/function.py:1]: I'm curious if this essentially does the same as https://github.com/aqueducthq/aqueduct/blob/main/src/python/aqueduct_executor/start-function-executor.sh and they should eventually merge

File: src/python/aqueduct_executor/operators/function_executor/extract_function.py [hunk 3]
```diff
@@ -52,7 +52,7 @@ def extract_function(storage: Storage, spec: FunctionSpec) -> None:
     """
     fn_path = spec.function_extract_path
     if not os.path.exists(fn_path):
-        os.mkdir(fn_path)
+        os.makedirs(fn_path)
```
Comment (cw75) [inline src/python/aqueduct_executor/operators/function_executor/extract_function.py:55]: Is there a difference between the two?

File: src/dockerfiles/lambda/connectors/athena.dockerfile (hunk 1 above)
Comment (cw75) [reply src/dockerfiles/lambda/connectors/athena.dockerfile:13]: Yes we should. @hsubbaraj-spiral could you hardcode the version number just like what we are doing for the k8s images and update the runbook to contain steps to build and push images? Thx!

**Review by cw75 — APPROVED** [review]
LGTM! Some comments inline.

File: src/python/aqueduct_executor/operators/function_executor/extract_function.py (hunk 3 above)
Comment (hsubbaraj-spiral) [reply src/python/aqueduct_executor/operators/function_executor/extract_function.py:55]: Apparently os.mkdir doesn't work on lambda for some reason, the suggestion is to use os.makedirs

File: src/dockerfiles/lambda/connectors/athena.dockerfile (hunk 1 above)
Comment (hsubbaraj-spiral) [reply src/dockerfiles/lambda/connectors/athena.dockerfile:13]: Yep, I'll change this before merging to main, just want to be able to test the whole branch together before.

**Pull request merged**